lua/auto-dark-mode/init.lua

```diff
@@ -86,25 +85,30 @@ local function init()
 	  ]])
 		end
 
-		query_command = table.concat({
+		query_command = vim.tbl_extend("keep", {
 			"dbus-send --session --print-reply=literal --reply-timeout=1000",
 			"--dest=org.freedesktop.portal.Desktop",
 			"/org/freedesktop/portal/desktop",
 			"org.freedesktop.portal.Settings.Read",
 			"string:'org.freedesktop.appearance'",
 			"string:'color-scheme'",
-		}, " ")
+		})
 	elseif system == "Windows_NT" or system == "WSL" then
 		-- Don't swap the quotes; it breaks the code
-		query_command =
-			'reg.exe Query "HKCU\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Themes\\Personalize" /v AppsUseLightTheme | findstr.exe "AppsUseLightTheme"'
+		query_command = {
+			"reg.exe",
+			"Query",
+			"HKCU\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Themes\\Personalize",
+			"/v",
+			"AppsUseLightTheme",
+		}
 	else
 		return
 	end
 
 	if vim.fn.has("unix") ~= 0 then
 		if vim.loop.getuid() == 0 then
-			query_command = "su - $SUDO_USER -c " .. query_command
+			query_command = vim.tbl_extend("keep", { "su", "-", "$SUDO_USER", "-c" }, query_command)
```

Comment:
I'm unsure how this command will behave now.
It seems like `su` will spawn the default shell indicated in `/etc/passwd`?